code/compute_fc/compute_fc.py

```diff
@@ -0,0 +1,883 @@
+        logging.error(f'"{derivatives_name}" could not be found on path !')
+        return ""
+    return find_derivative(path=op.split(path)[0])
+
+
+def check_existing_output(output, func_filename, return_existing=False, **kwargs):
+    logging.debug("\n\t".join(func_filename))
+
+    missing_data_filter = [
+        not op.exists(op.join(output, get_bids_savename(filename, **kwargs)))
+        for filename in func_filename
+    ]
+
+    missing_data = np.array(func_filename)[missing_data_filter]
+    logging.debug(f"\t{sum(missing_data_filter)} missing data found")
+    logging.debug("\n\t".join(missing_data))
+
+    if return_existing:
+        existing_data = np.array(func_filename)[
+            [not fltr for fltr in missing_data_filter]
+        ]
+        return missing_data.tolist(), existing_data.tolist()
+    return missing_data.tolist()
+
+
+def load_timeseries(func_filename, output):
+    if len(func_filename):
+        logging.info(f"Loading existing timeseries for {len(func_filename)} files ...")
+
+    loaded_ts = []
+    for filename in func_filename:
+        path_to_ts = get_bids_savename(
+            filename, patterns=TIMESERIES_PATTERN, **TIMESERIES_FILLS
+        )
+        logging.debug(f"\t{op.join(output, path_to_ts)}")
+        loaded_ts.append(
+            np.genfromtxt(op.join(output, path_to_ts), float, delimiter="\t")
+        )
+
+    return loaded_ts
+
+
+def get_confounds_manually(func_filename, **kwargs):
+    confounds, sample_mask = [], []
+
+    for filename in func_filename:
+        dir_name = op.dirname(filename)
+        confounds_file = op.join(
+            dir_name,
+            get_bids_savename(filename, patterns=CONFOUND_PATTERN, **CONFOUND_FILLS),
+        )
+
+        # confounds_json_file = load_confounds._get_json(confounds_file)
+        confounds_json_file = confounds_file.replace("tsv", "json")
+        individual_sm, individual_conf = _load_single_confounds_file(
+            confounds_file=confounds_file,
+            confounds_json_file=confounds_json_file,
+            **kwargs,
+        )
+        confounds.append(individual_conf)
+        sample_mask.append(individual_sm)
+
+    return confounds, sample_mask
+
+
+def fit_transform_patched(
+    func_filename, atlas_filename, confounds=None, sample_mask=None, **kwargs
+):
+    masker = MultiNiftiMapsMasker(maps_img=atlas_filename, **kwargs)
+
+    try:
+        time_series = masker.fit_transform(
+            func_filename, confounds=confounds, sample_mask=sample_mask
+        )
+    except ValueError:
+        # See nilearn issue #3967 for more details
+        logging.warning("Using patched version of 'MultiNiftiMapsMasker ...'")
+        masker = MultiNiftiMapsMasker_patched(maps_img=atlas_filename, **kwargs)
+
+        time_series = masker.fit_transform(
+            func_filename, confounds=confounds, sample_mask=sample_mask
+        )
+
+    return time_series
+
+
+def interpolate_and_denoise_timeseries(
```

Comment:
Did you check that we are doing what we described in the pre-registration? (I'm a bit lost so it might totally be; it is just hard for me to tell)

> FC will be extracted from whole-brain, RSfMRI time-series generated with fMRIPrep, sampled on the cortical mid-thickness surface, and on a regular, volumetric grid for the subcortical and cerebellar regions and stored in a mixed volume and surface format (CIFTI2). Time points at the onset of large head motions will be removed and replaced by interpolation from contiguous time points using cubic B-Spline kernels, whenever the corresponding framewise displacement (Power et al., 2014) calculated by fMRIPrep is above 0.4 mm. A subset of the confounds produced by fMRIPrep as well as temporal discrete cosine regressors, will be projected on the orthogonal of the signal space (Friston et al., 1994) and simultaneously (Lindquist et al., 2019) removed from the mixed volume and surface (CIFTI2) time series